lib/matplotlib/colorbar.py

```diff
@@ -577,30 +533,19 @@ def _get_ticker_locator_formatter(self):
         """
         locator = self.locator
         formatter = self.formatter
-        if locator is None:
-            if self.boundaries is None:
-                if isinstance(self.norm, colors.NoNorm):
-                    nv = len(self._values)
-                    base = 1 + int(nv / 10)
-                    locator = ticker.IndexLocator(base=base, offset=0)
-                elif isinstance(self.norm, colors.BoundaryNorm):
-                    b = self.norm.boundaries
-                    locator = ticker.FixedLocator(b, nbins=10)
-                elif isinstance(self.norm, colors.LogNorm):
-                    locator = _ColorbarLogLocator(self)
-                elif isinstance(self.norm, colors.SymLogNorm):
-                    # The subs setting here should be replaced
-                    # by logic in the locator.
-                    locator = ticker.SymmetricalLogLocator(
-                                      subs=np.arange(1, 10),
-                                      linthresh=self.norm.linthresh,
-                                      base=10)
-                else:
-                    if mpl.rcParams['_internal.classic_mode']:
```

Comment:
I'm biting the bullet and just removing this.  This changes 10 of the image tests below and a few of the fixed values in `test_colorbar`, but otherwise has no ill effect.  Its hard to wrap this as a special case just for linear scales/norms, and I don't see any reason to do this.